src/main/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtil.java

```diff
@@ -74,7 +74,8 @@ public static Scope getSurroundingScope(DetailAST node) {
                 final DetailAST mods =
                     token.findFirstToken(TokenTypes.MODIFIERS);
                 final Scope modScope = getScopeFromMods(mods);
-                if (returnValue == null || returnValue.isIn(modScope)) {
+                if (returnValue == null || returnValue.isIn(modScope)
+                        || isSurroundedWithinHigherScope(returnValue, modScope)) {
```

Comment:
> Diff Regression config: https://raw.githubusercontent.com/AkMo3/akmo3.github.io/master/ScopeChange/config-ScopeChange-20210417113718.xml

You have to be extremely careful when you change utility methods. Since they are not checks, you cannot perform regression on just them. Utility methods can be used by any number of checks, and each check impacted would have to be verified to ensure there is no negative impact.

The changes to tests show one part on those that you have touched on in many other checks. You need to include them all in regression. If you tests have changes, I am expecting those same checks to show up in the differences report.

You should also extend your view to any other check using this util but has not reported differences.

Reply:
Yes, I will be creating reports here so that everything remains transparent. Just first waiting to confirm that logic is correct.

Reply:
Sir, kindly confirm the new violations in the report.

Reply:
I will review the violations in the report. You should be looking at them first when they become available.